src/inmanta/agent/agent.py

```diff
@@ -1223,7 +1198,7 @@ async def get_code(
             - collection of ResourceInstallSpec for resource_types with valid handler code and pip config
             - set of invalid resource_types (no handler code and/or invalid pip config)
         """
-        if self._loader is None:
+        if not self.executor_manager.can_load_code():
```

Comment:
I have a slight preference for making this a field `_code_loader: True` on `self`, so that we don't have to pollute the generic executor interface with a testing artifact specific to the in-process executor.

But I'm not opposed to how you did it, if you prefer it like that.

Reply:
I think this is a lot cleaner with `_code_loader` indeed :+1: